src/sql/platform/accounts/common/accountViewModel.ts

```diff
@@ -48,23 +53,64 @@ export class AccountViewModel {
 		// 2) For each provider, get the accounts
 		// 3) Build parameters to add a provider and return it
 		try {
-			const metadata = await this._accountManagementService.getAccountProviderMetadata();
-			const accounts = await Promise.all(metadata.map(async providerMetadata => {
-				try {
-					const accounts = await this._accountManagementService.getAccountsForProvider(providerMetadata.id);
-					return <AccountProviderAddedEventParams>{
-						addedProvider: providerMetadata,
-						initialAccounts: accounts
-					};
-				} catch (err) {
-					this._logService.warn(`Error getting accounts for provider ${providerMetadata.id} : ${err}`);
-					return undefined;
-				}
-			}));
+			let accounts = await this.getAccountsForProviders();
+			if (this._authenticationService) {
+				const sessionAccounts = await this.getAccountForSessions();
+				accounts = accounts.concat(sessionAccounts);
+			}
+
 			return coalesce(accounts);
 		} catch (err) {
 			this._logService.warn(`Error getting account provider metadata : ${err}`);
 			return [];
 		}
 	}
+
+	private async getAccountsForProviders(): Promise<AccountProviderAddedEventParams[]> {
+		const metadata = await this._accountManagementService.getAccountProviderMetadata();
+		const accounts = await Promise.all(metadata.map(async (providerMetadata) => {
+			try {
+				const accounts = await this._accountManagementService.getAccountsForProvider(providerMetadata.id);
+
+				return <AccountProviderAddedEventParams>{
+					addedProvider: providerMetadata,
+					initialAccounts: accounts
+				};
+			} catch (err) {
+				this._logService.warn(`Error getting accounts for provider ${providerMetadata.id} : ${err}`);
+				return undefined;
```

Comment:
The return type isn't accurate with this - it should be `Promise<(AccountProviderAddedEventParams | undefined)[]>`

That being said, coalescing the results before returning wouldn't be a bad idea just to simplify this a bit